### DISCUSSION TITLE
UEFI BIOS flash over ipxe failing due to the flash utility being read only

### DISCUSSION THREAD

**Discussion by alan-abmxservers** [ipxe/ipxe#941] [Q&A]
Hi,
I'm having a bit of an issue with flashing BIOS files over iPXE.
Currently the flashing utility fails because it is loaded as read-only. 

The config file option looks like this:
:X12SPL-F
initrd http://${server_ip}/bios/supermicro/x12spl-f/bios.bin
initrd http://${server_ip}/bios/supermicro/x12spl-f/SUM.smc
initrd http://${server_ip}/bios/supermicro/x12spl-f/flash.nsh
initrd http://${server_ip}/bios/supermicro/x12spl-f/startup.nsh
chain http://${server_ip}/shell/Shell.efi
boot

I'm able to work around this issue by serving clonezilla over ipxe and running a flash utility from from the debian shell there, however this is much more time consuming and not ideal, and I'd prefer just flashing the BIOS through a UEFI shell if possible. 

This works locally when run from USB in a UEFI shell, and the SUM tool runs fine from within a linux environment.
I made sure to give sufficient permissions and executable access to the directory and files on the ipxe server itself, so I'm assuming it's because it's loaded via initrd?

Is there a better or alternative way than initrd to load the tool to the same "disk" as the other files that are being fetched which would allow the tool to have write access? 

I'm essentially having the same issue as the one posted here, and I've yet to find a solution: 
https://forum.ipxe.org/archive/index.php?thread-17479.html

Any advice would be greatly appreciated.

**Comment by NiKiZe** [comment]
Load shell.efi as bootx64.efi using the -n parameter.
Then chain into wimboot.

What this does is that it creates a virtual vfat filesystem that the files will be available over.

This is a hack, but it hopefully works for you.

**Reply by alan-abmxservers** [reply]
> Load shell.efi as bootx64.efi using the -n parameter. Then chain into wimboot.
> 
> What this does is that it creates a virtual vfat filesystem that the files will be available over.
> 
> This is a hack, but it hopefully works for you.

Thanks for this suggestion, I'll give it a try.
